src/keymap-priv.c

```diff
@@ -31,18 +31,29 @@
 static void
 update_builtin_keymap_fields(struct xkb_keymap *keymap)
 {
-    /* Predefined (AKA real, core, X11) modifiers. The order is important! */
+    /* Predefined (AKA real) modifiers. The order is important!
+     * Indeed, the modifier position defines the bit used to encode it.
+     * We use explicit indexes here to facilitate debugging. */
     static const char *const builtin_mods[] = {
+        /* Core/X11 modifiers */
         [0] = "Shift",
         [1] = "Lock",
         [2] = "Control",
         [3] = "Mod1",
         [4] = "Mod2",
         [5] = "Mod3",
         [6] = "Mod4",
-        [7] = "Mod5"
+        [7] = "Mod5",
+        /* xkbcommon extension */
+        [8] = "Mod6",
```

Comment:
Is it worth naming those `HMod1` through to `HMod5`? For "high modifier", or whatever other name we pick, just to make sure these mods are different to `Mod1`-`5`?

Reply:
Continuing the sequence seems more natural to me.

If we were to start a new sequence, then I suggest we use a longer prefix, e.g. `HigherMod1`, in order to make clear it is not a typo. I am not sure “higher” is a good description though. Maybe “extended” is better, such as `ExtMod1` or `ModExt1`?

Reply:
yeah, naming things is hard. Either way, I'm not going to fight hard for this so unless we come up with some outstanding name, `Mod6` and friends it is.